crates/gossip/src/respond/initiate.rs

```diff
@@ -593,6 +590,103 @@ mod tests {
         }
     }
 
+    #[tokio::test]
+    async fn resolve_tie_break_tie() {
```

Comment:
```suggestion
    async fn tie_breaker_conflict_cancels_gossip_session() {
```

The flow here is that both sides initiate at the same time with the same tie breaker. Then our side spots that our tie_breaker is *not* greater than the other side's tie_breaker and so we drop our initiate and send back an accept. We can actually expect that the other side would do the same thing so this test is really saying that if both sides send an initiate, at the same time, with the same tie_breaker, then both sides will cancel their session and try to accept the other one which is likely going to result in the session failing to initiate either way right?

Reply:
I can change the name; I was trying to keep it consistent with the other two tests `resolve_tie_break_win` and `resolve_tie_break_lose`.

As for the logic then yeah, it's all very timing-specific and incredibly unlikely that two agents will initiate at the same time **and** choose the same random number for the `tie_breaker` which is why I'm not too worried about "fixing" it and just created #316 just so that we're aware. I do think it's something that we should test, though, so we can hopefully catch if the logic changes.

If it does happen, then I think that both agents will accept and wait for the other agent to do something, and they will remain in this state until the gossip-round timeout elapses, which is set to 15 seconds, then they will try again.